include/s3select.h

```diff
@@ -1816,17 +1816,18 @@ class csv_object : public base_s3object
     {
       do
       {
-
         number_of_tokens = getNextRow();
         if (number_of_tokens < 0) //end of stream
         {
           if (m_is_to_aggregate)
-            for (auto& i : m_projections)
+            for (auto iter = m_projections.begin(); iter != m_projections.end(); iter++)
             {
-              i->set_last_call();
-              i->set_skip_non_aggregate(false);//projection column is set to be runnable
-              result.append( i->eval().to_string() );
-              result.append(",");
+              (*iter)->set_last_call();
+              (*iter)->set_skip_non_aggregate(false);//projection column is set to be runnable
+              result.append( (*iter)->eval().to_string() );
+              if ( iter != m_projections.end() - 1 ) {
+                result.append(1u, m_csv_defintion.column_delimiter);
```

Comment:
m_csv_definition is the correct object to hold output-serialization info.
m_csv_defintion.column_delimiter contains the input-column-delimiter